lms/djangoapps/teams/static/teams/js/spec/edit_team_spec.js

```diff
@@ -0,0 +1,128 @@
+                "id": "tn",
+                "name": "TeamName",
+                "is_active": true,
+                "course_id": "a/b/c",
+                "topic_id": "awesomeness",
+                "date_created": "2015-07-29T09:59:37.528Z",
+                "description": "TeamDescription",
+                "country": "c",
+                "language": "a",
+                "membership": []
+            });
+
+            expect(teamEditView.$('.wrapper-msg').is(':visible')).toBeFalsy();
+            expect(Backbone.history.navigate.calls[0].args).toContain('topics/awesomeness');
+        });
+
+        it('shows validation error message when field is empty', function () {
+            verifyValidation([
+                ['.u-field-name input', 'Name'],
+                ['.u-field-textarea textarea', '']
+            ]);
+            teamEditView.$('.wrapper-msg').hide();
+            verifyValidation([
+                ['.u-field-name input', ''],
+                ['.u-field-textarea textarea', 'description']
+            ]);
+            teamEditView.$('.wrapper-msg').hide();
+            verifyValidation([
+                ['.u-field-name input', ''],
+                ['.u-field-textarea textarea', '']
+            ]);
+        });
+
+        it('shows validation error message when field value length exceeded the limit', function () {
+            var teamName = new Array(500 + 1).join( '$' );
+            var teamDescription = new Array(500 + 1).join( '$' );
+
+            verifyValidation([
+                ['.u-field-name input', teamName],
+                ['.u-field-textarea textarea', 'description']
+            ]);
+            teamEditView.$('.wrapper-msg').hide();
+            verifyValidation([
+                ['.u-field-name input', 'name'],
+                ['.u-field-textarea textarea', teamDescription]
+            ]);
+            teamEditView.$('.wrapper-msg').hide();
+            verifyValidation([
+                ['.u-field-name input', teamName],
+                ['.u-field-textarea textarea', teamDescription]
+            ]);
+        });
+
+        it("shows an error message for HTTP 500", function () {
+            var requests = AjaxHelpers.requests(this);
+
+            teamEditView.$('.u-field-name input').val('TeamName');
+            teamEditView.$('.u-field-textarea textarea').val('TeamDescription');
+
+            teamEditView.$('.create-team.form-actions .action-primary').click();
+            AjaxHelpers.respondWithError(requests);
+
+            expect(teamEditView.$('.wrapper-msg .copy').text().trim()).toBe("An error occurred. Please try again.");
+        });
+
+        it("changes route on cancel click", function () {
+            teamEditView.$('.create-team.form-actions .action-cancel').click();
+            expect(Backbone.history.navigate.calls[0].args).toContain('topics/awesomeness');
+        });
+
+        it("only highlight fields with errors", function () {
```

Comment:
See my comments above - I think the logic in this test case should be handled when we call `verifyValidation`.